File: docs/examples/methodFunctions/onEventTick.ts
import { LightVM, VMEvent } from 'lightvm';

const vm = new LightVM({ caps: [] });

vm.on(VMEvent.Tick, (data) => {
  console.log('Event: ', data.event);
  console.log('Payload: ', data.payload);
});
vm.run();

File: docs/examples/methodFunctions/on_event_tick.rs
use lightvm::LightVM;
use lightvm::types::{vmconfig::VmConfig, vmevent::VmEvent};

fn main() {
  let mut vm = LightVM::new(VmConfig {
    caps: vec![],
    ..Default::default()
  });

  vm.on(VmEvent::Tick, |data| {
    println!("Event: {:?}", data.event);
    println!("Payload: {:?}", data.payload);
  });
  vm.run(None);
}
